src/OnChain.hs

```diff
@@ -36,18 +36,22 @@ import qualified PlutusTx.Builtins        as BI
 import           Types
 
 
+-- TODO What is the difference between ENOPNFT and ENNFT? I assumed that ENNFT
+-- is preminted and distributed to the EN operators. Then, was is ENOPNFT used
+-- for?
+
```

Comment:
Correct ENNFT is preminted and distributed. 
ENOP-NFT is minted in the registration transaction and is needed to de-register  (ENOP NFT burnt and ENNFT unlocked) or modify (ENOP NFT spent in this tx) the registration. 
Without the ENOP NFT the registration would be bound to a wallet/address but we want for example that an EN can be traded when it is operational.